### DISCUSSION TITLE
How to show nchat theme colors inside a tmux session?

### DISCUSSION THREAD

**Discussion by HawkingRadiation42** [d99kris/nchat#116] [Q&A]
as the title describes, i have kept tokyo nights as mytheme but when using tmux and then using nchat withing the session then nchat themes doesnt work, any idea for it?

**Comment by d99kris** [comment]
Hi @HawkingRadiation42 - this question came up in the [telegram group](https://t.me/nchatusers) as well recently. 

What output do you get in your terminal (not running tmux) from the following command?
`echo $TERM`

Assuming it's `xterm-256color`, can you try a workaround to run nchat in tmux with that TERM set and see if it works? Example:
`TERM=xterm-256color nchat`

Also, can I check what terminal emulator/app you're using?

The root cause is probably a bug in nchat (assuming other terminal apps works well in tmux), but I'll need to do some research to see how it can be fixed.

**Reply by HawkingRadiation42** [reply]
hello, yeah its xterm-256color and im using iterm2.
edit: yeah by setting the TERM variable to xterm-256color does work thankyou

**Reply by HawkingRadiation42** [reply]
for users facing this error, you can using this command "TERM=xterm-256color nchat"
and for simplicity can add alias of this for nchat so by default opens in this. hope it helps someone

**Reply by d99kris** [reply]
For reference - there's an issue #141 reported, to see if it can be handled by nchat.

**Comment by Disonantemus** [comment]
For reference, I've got no colors with [st (virtual terminal emulator)](https://st.suckless.org), I think is the same problem, because my TERM is different:
~~~
$ echo $TERM
screen-256color
~~~
If I do:
~~~
TERM=xterm-256color nchat
~~~
Works as expected, with colors, in st, tmux, TTY. Then is not only a **tmux** problem, is for:
- every terminal emulator.
- every multiplexer.
- TTY.

that has a different TERM.

---

In README only says tmux, maybe you can change that, 4 people with same problem and not using tmux.

**Reply by d99kris** [reply]
Thanks for sharing another data point 👍 

Admittedly I don't know the exact scope of the issue, but I don't think it's affecting everything "that has a different TERM", but rather using a `TERM` which is not present in the system's `terminfo` database (or the `TERM` profile being present in the `terminfo` db but it's wrongly set up without the custom colors capability `ccc`).

In any case, given that you see the issue in `st` without tmux I agree a README update would be useful. Perhaps the FAQ entry can be broadened to something like:
```
Q: Custom colors are not shown when running nchat?

A: Try run nchat with a TERM supporting custom colors, e.g. `TERM=xterm-256color nchat`.
```

I'll try make a change next time I update the readme.

**Reply by d99kris** [reply]
Btw, just for my curiosity - when you say `I've got no colors with st` - do you mean a plain `st` terminal? 

And the `TERM` value `screen-256color` you shared is for `st` (not with `screen` or `tmux` running)?

**Reply by Disonantemus** [reply]
- Yes both.
- I did some checks, and I my `.bashrc` has `export TERM=screen-256color` that can explain is not your expected value, that only depends on bash and because of that, is going to affect every terminal emulator (not only st).
- I don't remember why is there, but maybe was looking 4 ways to show colors in TTY.

**Reply by d99kris** [reply]
Thanks for sharing!
